### DISCUSSION TITLE
HTML Hyperlinks in PDF

### DISCUSSION THREAD

**Discussion by hmennen90** [spatie/browsershot#646] [Q&A]
Does Browsershot make HTML Hyperlinks usable in PDF Files?

**Comment by babekasadli** [comment]
This happens sometimes and sometimes not. Text and images can be copied when the web page is converted to PDF format. This means that this may vary depending on the processing position of the hyperlink tag `<a>`. Because these tags can often be used inside a `<div>` block, in a list or in a paragraph. If you want to add links to a pdf file later, you can add links by editing the PDF file. For free PDF editing, I can recommend you https://www.sejda.com/pdf-editor service. The advantage of this service is that you can add links not only on the text but also on the images

**Reply by hmennen90** [reply]
Thanks for your reply - but in our case, we Use Data provided by Shopware and load them with Hyperlinks in a HTML. This HTML is loaded in Browsershot for generating a PDF.

**Reply by jbostoen** [reply]
Did you find a solution to this? It seems that relative links are also converted to full links in the document?

Aside from that, I was also hoping to generate a table of contents (TOC)/outline in the PDF document, but it seems it's not possible?

**Reply by ChrisSantiago82** [reply]
I have the following issue:

This doesn't create a link
`<a href="https://google.com">test</a>`

This does
`<a href="https://google.com">https://google.com</a>`

But of course, this doesn't look nice. Is there a parameter or something that can be used?
